tests/samplers_tests/test_samplers.py

```diff
@@ -53,9 +53,15 @@
         []
         if sys.version_info < (3, 7, 0)
         else [
-            lambda: optuna.integration.BoTorchSampler(n_startup_trials=0),
             lambda: optuna.samplers.QMCSampler(),
         ]
+    )
+    # TODO(nzw0301): Remove version constraints if BoTorch supports Python 3.10
+    # or Optuna does not support Python 3.6.
+    + (
+        []
+        if sys.version_info >= (3, 10, 0) or sys.version_info < (3, 7, 0)
+        else [lambda: optuna.integration.BoTorchSampler(n_startup_trials=0)]
```

Comment:
I am not against this solution, but how will it remind us later?

Reply:
As you can see, this PR adds to-do lines to L59 -- L60 in this file. If you mean that the todo is not enough, I would like to add explicit tasks to https://github.com/optuna/optuna/issues/2981 for the followups.

Reply:
Thanks! That would be a good solution!

> I would like to add explicit tasks to https://github.com/optuna/optuna/issues/2981 for the followups.